components/layout/construct.rs

```diff
@@ -1341,15 +1328,9 @@ impl<'a> PostorderNodeMutTraversal for FlowConstructor<'a> {
             // TODO(pcwalton): Make this only trigger for blocks and handle the other `display`
             // properties separately.
 
-            (_, float::T::none, _) => {
-                let construction_result = self.build_flow_for_nonfloated_block(node);
-                self.set_flow_construction_result(node, construction_result)
-            }
-
-            // Floated flows contribute float flow construction results.
             (_, float_value, _) => {
                 let float_kind = FloatKind::from_property(float_value);
-                let construction_result = self.build_flow_for_floated_block(node, float_kind);
+                let construction_result = self.build_flow_for_block(node, float_kind);
```

Comment:
The [property definition tables](http://dev.w3.org/csswg/css-multicol/#the-number-and-width-of-columns) of the `column-count` and `column-width` properties contain:
> Applies to: non-replaced block-level elements (except table elements), table cells, and inline-block elements 

Which, in a CSS 2.1 word, is really just a long way of saying “Applies to [block containers](http://dev.w3.org/csswg/css2/visuren.html#block-container-box)”.

In other words, it’s only after all the other box construction decisions have been made that we look at block containers and decide if they should be multicol instead.

By the way, I believe this `match` should only be based on `display`, not on `float` or `position`. The computed value (as set in the `style` crate) of `display` is already [affected `float` and `position`](http://dev.w3.org/csswg/css2/visuren.html#dis-pos-flo).